### PR TITLE
Feat/submarine exploration result reporter

### PR DESCRIPTION
**Pull request by shambarick**
Add a reporter for submarine voyage log to track loots.

### CODE REVIEW

File: apps/client/src/app/core/data-reporting/exploration-result.reporter.ts
Comment (Supamiu) [inline apps/client/src/app/core/data-reporting/exploration-result.reporter.ts:4]: Even if it may seem a bit too much, one file per export is a general rule i'm trying to follow (well, sometimes I don't, but it's rare), I'd love this to be in its own file.

File: apps/client/src/app/core/data-reporting/submarine-exploration-result-reporter.ts
Comment (Supamiu) [inline apps/client/src/app/core/data-reporting/submarine-exploration-result-reporter.ts:11]: Could be nice having proper typings for these, they're implicit any and it makes the code harder to read and maintain.

File: apps/data-extraction/src/extractors/submarine-ranks.extractor.ts
Comment (Supamiu) [inline apps/data-extraction/src/extractors/submarine-ranks.extractor.ts:16]: Whenever it's possible, prefer a custom structure so if tomorrow things are renamed for whatever reason, only this part of the code will have to change. This also helps with naming convention as TC uses lowercase for property names while XIVAPI uses UpperCamelCase.

File: apps/data-extraction/src/extractors/submarine-parts.extractor.ts
Comment (Supamiu) [inline apps/data-extraction/src/extractors/submarine-parts.extractor.ts:34]: Having lazy files follow the lowercase naming convention would be nice, also same comment as for the other exporter regarding custom structures instead ^^.

File: apps/data-extraction/src/extractors/submarine-ranks.extractor.ts
Comment (shambarick) [reply apps/data-extraction/src/extractors/submarine-ranks.extractor.ts:16]: Np. Isn't there something that can be used for checking the struct so the extractor throws an error if the response doesn't contain the columns? Like an endpoint in XIVAPI that returns the body spec or just the list of the fields for a given index (eg: SubmarineRanks)
That would avoid the trouble of checking if there is struct changes manually every update.

File: apps/data-extraction/src/extractors/submarine-parts.extractor.ts
Comment (shambarick) [reply apps/data-extraction/src/extractors/submarine-parts.extractor.ts:34]: Yep, I noticed that a bit too late and I already pushed the code, was gonna fix that for the next push.

File: apps/data-extraction/src/extractors/submarine-ranks.extractor.ts
Comment (Supamiu) [reply apps/data-extraction/src/extractors/submarine-ranks.extractor.ts:16]: I'm the maintainer of XIVAPI so I'll be aware of changes since I'm the one making them 😉 to get the fields for a given sheet, just get one of the items, like https://xivapi.com/SubmarineRank/1, there's no optional fields, so every single field can be found on every single entry 👍

File: apps/client/src/app/core/data-reporting/exploration-result.reporter.ts
Comment (shambarick) [reply apps/client/src/app/core/data-reporting/exploration-result.reporter.ts]: For now, I have put it inside `model/other`. Might actually be better to move it in a `airship/submarine` module tho.

**Comment by shambarick** [comment]
@Supamiu , should be good now

File: apps/data-extraction/src/extractors/submarine-parts.extractor.ts
Comment (Supamiu) [inline apps/data-extraction/src/extractors/submarine-parts.extractor.ts:15]: You should use filters instead, this is the query string fragment for it: `filters=FilterGroup=36`, matching a number using a fulltext search is not the way to go for this one :)

File: apps/data-extraction/src/main.ts
Comment (Supamiu) [inline apps/data-extraction/src/main.ts:130]: The extractors are listed in order to avoid having issues with extractor dependencies. The mappy and LGB extractors require informations from previous extractors, which is why the array isn't sorted programatically, it's made by hand.

**Review by Supamiu — CHANGES_REQUESTED** [review]
I see you implemented a reporter but for submarine only, are you planning the airship one for later on or should I wait for it to merge?

File: apps/data-extraction/src/main.ts
Comment (shambarick) [reply apps/data-extraction/src/main.ts]: Sorry, I didn't know :/

**Comment by shambarick** [comment]
You can merge as is.

The airship one is for later because I still need a few more samples for the struct I'm missing. So I have no ETA for that one atm.

**Review by Supamiu — APPROVED** [review]

**Pull request merged**
